src/jit/gentree.cpp

```diff
@@ -8826,6 +8830,14 @@ unsigned GenTree::NumChildren()
         {
             return 3;
         }
+#if FEATURE_HW_INTRINSICS
+        // TODO-XArch-CQ: Check for possibility of setting SetZeroVector128 or similar intrinsics to unary operator
+        // format
+        if ((OperGet() == GT_HWIntrinsic) && (AsHWIntrinsic()->gtHWIntrinsicId == NI_SSE2_SetZeroVector128))
```

Comment:
We need to figure out why SSE_SetZeroVector128 works fine but SSE2_SetZeroVector128 not. Could you please take a look at JITDump? If we need this code finally, it should be more general.

Reply:
Agree, but can we do it in next PRs, there are 2 still coming from me for Sse2 only and #15777 is much more complex and perhaps best place to work on improving compiler handling of not only SSE2_SetZeroVector128 but also other complex intrinsic. The PR cycle is relatively heavy with CI tests and local test and it is more convenient to handle smaller changes together with other work.

My assumption is that `SSE2_SetZeroVector128`  probaly works the same for floating point based types but may diverge for integrals - this is the only significant difference with `SSE_SetZeroVector128`.

Reply:
If you accept to dely it I will open issue to track that work.

Reply:
It would be good to have a basic understanding of why it is needed before merging. IIRC this had something to do with loop hosting and might exist for `SSE.SetZeroVector128` (or any other 0/1 operand intrinsics) as well.

We should validate the reason and update the above code paths to handle them.

Reply:
Working on it. Will post details later.